iaso/utils/jsonlogic.py

```diff
@@ -36,28 +36,34 @@ def as_sql(self, compiler, connection):
 JSONField.register_lookup(ExtractForceFloat)
 
 
-def jsonlogic_to_q(jsonlogic: Dict[str, Any], field_prefix: str = "") -> Q:
+def jsonlogic_to_q(
+    jsonlogic: Dict[str, Any],
+    field_prefix: str = "",
+    recursion_func: Callable = None,
+) -> Q:
     """Converts a JsonLogic query to a Django Q object.
 
     :param jsonlogic: The JsonLogic query to convert, stored in a Python dict. Example: {"and": [{"==": [{"var": "gender"}, "F"]}, {"<": [{"var": "age"}, 25]}]}
     :param field_prefix: A prefix to add to all fields in the generated query. Useful to follow a relationship or to dig in a JSONField
+    :param recursion_func: Optionally specify a function to call for recursion, allowing this method to be "wrapped". By default, when no function is specified, it calls itself.
 
     :return: A Django Q object.
     """
 
+    func = jsonlogic_to_q if recursion_func is None else recursion_func
+
     if "and" in jsonlogic:
         return reduce(
             operator.and_,
-            (jsonlogic_to_q(subquery, field_prefix) for subquery in jsonlogic["and"]),
+            (func(subquery, field_prefix) for subquery in jsonlogic["and"]),
         )
```

Comment:
I know that was implemented before and this is largely a personal preference but… every time I see a `reduce()` call, I need to go read the doc to understand what's actually happening.

I tend to think that an accumulation loop is easier to understand, e.g.

```python
sub_query = Q()
for lookup in jsonlogic["and"]:
    sub_query = operator.and_(sub_query, func(lookup, field_prefix))
return sub_query
```